src/m365/spo/commands/file/file-roleinheritance-break.ts

```diff
@@ -0,0 +1,157 @@
+import commands from '../../commands';
+import * as SpoFileGetCommand from './file-get';
+import { Options as SpoFileGetCommandOptions } from './file-get';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  fileUrl?: string;
+  fileId?: string;
+  clearExistingPermissions?: boolean;
+  confirm?: boolean;
+}
+
+class SpoFileRoleInheritanceBreakCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FILE_ROLEINHERITANCE_BREAK;
+  }
+
+  public get description(): string {
+    return 'Breaks inheritance of file. Keeping existing permissions is the default behavior.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        fileUrl: typeof args.options.fileUrl !== 'undefined',
+        fileId: typeof args.options.fileId !== 'undefined',
+        clearExistingPermissions: !!args.options.clearExistingPermissions,
+        confirm: !!args.options.confirm
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '--fileUrl [fileUrl]'
+      },
+      {
+        option: 'i, --fileId [fileId]'
+      },
+      {
+        option: '-c, --clearExistingPermissions'
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.fileId && !validation.isValidGuid(args.options.fileId)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(['fileId', 'fileUrl']);
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const breakFileRoleInheritance: () => Promise<void> = async (): Promise<void> => {
+      try {
+        const fileURL: string = await this.getFileURL(args);
+
+        let keepExistingPermissions: boolean = true;
+        if (args.options.clearExistingPermissions) {
+          keepExistingPermissions = !args.options.clearExistingPermissions;
+        }
+
+        const requestOptions: AxiosRequestConfig = {
+          url: `${args.options.webUrl}/_api/web/GetFileByServerRelativeUrl('${formatting.encodeQueryParameter(fileURL)}')/ListItemAllFields/breakroleinheritance(${keepExistingPermissions})`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
```

Comment:
We should remove the odata header for this command. Otherwise the error message will be in XML instead of JSON.

```suggestion
            accept: 'application/json'
```

Reply:
not `application/json;odata=nometadata` 🤔?

Reply:
In hindsight, that's better yeah 😊

Reply:
then done properly I guess 🙂